frontends/web/src/routes/settings/electrum-servers.tsx

```diff
@@ -0,0 +1,113 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ElectrumServer } from './electrum-server';
+import { getConfig, getDefaultConfig, setConfig } from '../../api/backend';
+import { confirmation } from '../../components/confirm/Confirm';
+import style from './electrum.module.css';
+import A from '../../components/anchor/anchor';
+
+type Props = {
+  coin: 'btc' | 'tbtc' | 'ltc' | 'tltc';
+};
+
+interface IElectrumServer {
+  server: string;
+  tls: boolean;
+  pemCert: string;
+}
+
+export const ElectrumServers = ({
+  coin
+}: Props) => {
+  const { t } = useTranslation();
+  const [config, setConfigState] = useState<any>();
+  const loadConfig = () => {
+    getConfig().then(setConfigState);
+  };
+  useEffect(loadConfig, []);
+  if (config === undefined) {
+    return null;
+  }
+  const electrumServers: IElectrumServer[] = config.backend[coin].electrumServers;
+
+  const save = async (newElectrumServers: IElectrumServer[]) => {
+    const currentConfig = await getConfig();
+    currentConfig.backend[coin].electrumServers = newElectrumServers;
+    await setConfig(currentConfig);
+    setConfigState(currentConfig);
+  };
+
+  const onAdd = (server: IElectrumServer) => {
+    let newElectrumServers = electrumServers.slice();
+    newElectrumServers.push(server);
+    save(newElectrumServers);
+  };
+
+  const onRemove = (index: number) => {
+    let newElectrumServers = electrumServers.slice();
+    newElectrumServers.splice(index, 1);
+    save(newElectrumServers);
+  };
+
+  const resetToDefault = () => {
+    confirmation(t('settings.electrum.resetConfirm'), response => {
+      if (response) {
+        getDefaultConfig().then(config => {
+          save(config.backend[coin].electrumServers);
+        });
+      }
+    });
+  };
+
+  const onRemoveCb = (server: IElectrumServer, index: number) => (() => {
+    confirmation(t('settings.electrum.removeConfirm', { server: server.server }), confirmed => {
+      if (confirmed) {
+        onRemove(index);
+      }
+    });
+  });
+
+  return (
+    <div className={style.serversContainer}>
+      <div className="row">
+        <div className={['flex flex-row flex-between flex-items-center', style.titleContainer].join(' ')}>
+          <h4 className="subTitle m-none">{t('settings.electrum.servers')}</h4>
+          <A href="#" className={['labelLarge labelLink', style.resetLink].join(' ')} onClick={resetToDefault}>{t('settings.electrum.reset')}</A>
+        </div>
+        <ul className={style.servers}>
+          {
+            electrumServers.map((server, index) => (
+              <ElectrumServer
+                // @ts-ignore
```

Comment:
Hm, does this do anything? I can't see any warnings on my side without it.

Reply:
Removed